### PR TITLE
chore: split "dev dependencies" and real dependencies

### PR DESCRIPTION
**Pull request by zepatrik**
## Related issue

#612 

## Proposed changes

This adds a separate `go.mod` to the `.bin` directory for managing all the binaries required for development/build,
That way we can remove those "dev dependencies" from the package itself.

### CODE REVIEW

**Comment by zepatrik** [comment]
Also:
```shell
$ go list -m all | wc -l
539
$ git checkout master
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
[patrik@patrik-x1 keto]$ go list -m all | wc -l
748
```

**Comment by aeneasr** [comment]
Without `go_mod_indirect_pins.go` `go mod tidy` will remove all dependencies when executed in `.bin`

File: .bin/go.mod [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+replace github.com/ory/kratos/corp => github.com/ory/kratos/corp v0.0.0-20210118092700-c2358be1e867
+
+replace github.com/oleiade/reflections => github.com/oleiade/reflections v1.0.1
+
+require (
+	github.com/bufbuild/buf v0.31.1
+	github.com/go-swagger/go-swagger v0.26.1
+	github.com/goreleaser/godownloader v0.1.1-0.20200426152203-fd8ad8f7dd78
+	github.com/mattn/goveralls v0.0.7
+	github.com/ory/cli v0.0.49
```
Comment (aeneasr) [inline .bin/go.mod:18]: Install ory/cli using curl bash and the need for any replaces is gone

**Comment by zepatrik** [comment]
> Without go_mod_indirect_pins.go go mod tidy will remove all dependencies when executed in .bin

That's why there is the indirect pins file? https://github.com/ory/keto/pull/619/files#diff-8bd5761c581528cf012968e778e74e9b9c9f219521da0ba7735e1fa5396893bd

File: .bin/go.mod (hunk 1 above)
Comment (zepatrik) [reply .bin/go.mod:18]: Yes, but we will run that more often than necessary. Also, all other go dependencies are still contributing to the dependencies:

> ```shell
> $ go list -m all | wc -l
> 539
> $ git checkout master
> Switched to branch 'master'
> Your branch is up to date with 'origin/master'.
> $ go list -m all | wc -l
> 748
> ```

This PR reduces the dependency count by ~28%.

**Comment by aeneasr** [comment]
Exactly!

File: .bin/go.mod (hunk 1 above)
Comment (aeneasr) [reply .bin/go.mod:18]: True, that's definitely nice!

**Review by aeneasr — APPROVED** [review]

**Pull request merged**